napari/layers/shapes/_shapes_utils.py

```diff
@@ -942,6 +942,9 @@ def points_in_poly(points, vertices):
         )
         cond_3 = np.logical_or(cond_1, cond_2)
         d = vertices[j] - vertices[i]
+        # Prevents floating point imprecision from generating false positives
+        tolerance = 1e-12
+        d = np.where(abs(d) < tolerance, 0, d)
```

Comment:
maybe we use `np.isclose(d, 0, atol=1e-12)` here instead - see https://numpy.org/doc/stable/reference/generated/numpy.isclose.html#numpy.isclose

Reply:
Hi @sofroniewn , no worries! I understand how many tasks a project this size demands!

Indeed, this case seems to be where `np.isclose` was meant to be used, good suggestion!
Since it returns a `bool` array, there is a need for something like this `d = np.where(np.isclose(d, 0, atol=1e-12), 0, d)` in order to work.

Reply:
that looks good to me. maybe @Carreau can advise and review this PR too - thanks!!

Reply:
That seem reasonable, I'm just  astonished that we roll our own algorithme, instead of using say something like https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.tsearch.html#scipy.spatial.tsearch in the long run.

The one above does the opposite (find simplex given point), but I'm astonished that there is not an existing function in scipy that given a point tell us whether or not it's in a polygon/triangle. For sure the above algrithm uses something like that. (but we can postpone a refactor to another time)

Reply:
> I wasn't aware of tsearch in scipy

Yeah, I had to look it up too. 

From https://stackoverflow.com/questions/36399381/whats-the-fastest-way-of-checking-if-a-point-is-inside-a-polygon-in-python there are a few variants that exists in non-scipy libraries, but I agree that there likely should be one in scipy.

Reply:
>  Maybe we'll get this PR merged and then create an issue to refactor this code to use more builtin scipy functions 

I can create the issue, I think that also a proper fix would be "point in shape", 
it's likely much easier to check if a point is in an ellipse from the ellipse corners, than to triangulate it and check if it's in the triangulation.

Reply:
Done as https://github.com/napari/napari/issues/4343 merging this in the meantime,